### ISSUE TITLE
Can I specify "mode": "SIGN_MODE_DIRECT"

### ISSUE THREAD

**Issue by SalletOne** [terra-money/terra.kt#3]
Can I specify "mode": "SIGN_MODE_DIRECT"

**Comment by jdekim43** [comment]
Not support currently.
It will be support after develop protobuf feature.

**Comment by jdekim43** [comment]
Maybe you can develop using `money.terra:proto-kotlin:0.5.16` and TerraWallet.key.sign() function.

```kotlin
val walletKey = TerraWallet.create().first.key as RawKey
val txBody = txBody {
    messages.add(...)
...
}
val authInfoObj = authInfo {
    signerInfos.add(signerInfo {
        publicKey = any {
            typeUrl = "/cosmos.crypto.secp256k1.PubKey"
            value = publicKey {
                secp256K1 = ByteString.copyFromUtf8(walletKey.publicKeyHex)
            }.toByteString()
        }
        sequence = 0
        modeInfo = modeInfo {
            single = single {
                mode = Signing.SignMode.SIGN_MODE_DIRECT
            }
        }
    })
}
val signDoc = signDoc {
    accountNumber = 0
    authInfoBytes = authInfoObj.toByteString()
    chainId = "bombay-12"
    bodyBytes = txBody.toByteString()
}
val signature = walletKey.signSync(signDoc.toByteArray())
val tx = tx {
    signatures.add(ByteString.copyFromUtf8(Base64.getEncoder().encodeToString(signature)))
    authInfo = authInfoObj
    body = txBody
}

//and broadcast tx object using grpc (money.terra:grpc-kotlin:0.5.16)
```

**Comment by SalletOne** [comment]
Hello, we are doing cold wallets here. I need to get useful code from your source code. If you import your sdk directly, it will conflict with our jar package. If we build with the existing code Transaction, will this SIGN_MODE_DIRECT expire in the future? In python, we found that this mode will expire in the future.


in python

class SignMode(betterproto.Enum):
    """SignMode represents a signing mode with its own security guarantees."""

    # SIGN_MODE_UNSPECIFIED specifies an unknown signing mode and will be
    # rejected
    SIGN_MODE_UNSPECIFIED = 0
    # SIGN_MODE_DIRECT specifies a signing mode which uses SignDoc and is
    # verified with raw bytes from Tx
    SIGN_MODE_DIRECT = 1
    # SIGN_MODE_TEXTUAL is a future signing mode that will verify some human-
    # readable textual representation on top of the binary representation from
    # SIGN_MODE_DIRECT
    SIGN_MODE_TEXTUAL = 2
    # SIGN_MODE_LEGACY_AMINO_JSON is a backwards compatibility mode which uses
    # Amino JSON and will be removed in the future
    SIGN_MODE_LEGACY_AMINO_JSON = 127

**Comment by jdekim43** [comment]
If you want to write code that can be used in the future, you should use SIGN_MODE_DIRECT .
Currently, terra.kt only supports SIGN_MODE_LEGACY_AMINO_JSON and is preparing to support SIGN_MODE_DIRECT with protobuf.
However, if you want to implement SIGN_MODE_DIRECT right away, you can implement it yourself by referring to the code written above.

And which package is conflict? Conflicts for org.bitcoinj are fixed in v0.20.5.
